src/posix/platform/openthread-core-posix-config.h

```diff
@@ -153,7 +153,7 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_ENTRIES
-#define OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_ENTRIES 32
+#define OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_ENTRIES 128
```

Comment:
How about changing `OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_MAX_SNOOP_ENTRIES` here as well?

Suggest using some value in ~25 range.
```
#define OPENTHREAD_CONFIG_TMF_ADDRESS_CACHE_MAX_SNOOP_ENTRIES  25
```

Basically in default we had 10 total entries, and allowed 2 of them to be used for snoop optimization. If we scale up the total number of entry to 128, we can allow ~20 for snoop (keeping ratio roughly same).

Snoop optimization can help reduce the number of address queries in general.

Reply:
The traffic pattern that matter is when the node sending a (request) message to some entity outside of Thread mesh and then expecting to receive a response from it. This all goes through a BR.

With snoop optimization, the BR will create a snoop cache entry when it sees the the request message and then does not need to perform address query when it gets the response to forward to the node. 

In the design, we limit the number snoop entries (which is the `ADDRESS_CACHE_MAX_SNOOP_ENTRIES`) to ensure snooped entries do not overwrite the other cache entries.  If a snooped entry is used to resolve, the entry become a normal cache entry.  We keep an snooped entry around (don't allow it to reclaimed) for a short time specified by `SNOOP_CACHE_ENTRY_TIMEOUT` to give a chance for it to be used.

Keeping the snoop entries to default of `2` will still work fine, but if we increase the total number of cache from 10 (or 32) to 128, it make sense to me to also scale up the number of snoop entries (to gain the benefit from it).

Here is how I would choose: If we think we want `x` cache entries in stead state (based on the use-case), we can use `1.2 * x` as the total and allow `0.2 * x` to be used as snoop entries.

------

This feature and other such optimizations (which are not really part of Thread spec but enhancement added in OpenThread implementation) typically originate from performance observations on the deployed Thread networks in our internal products/projects.

Reply:
seems this then become a question of how much un-acked traffic is travelling off the mesh. That's the only case that would create a potentially negative scenario from having the cache_snoop entries higher (i don't believe any of this matters if devices on mesh < max_cache).

I'd personally vote for that value to be based around updated testing on real world scenarios (particularly, with devices >128). Our testing at the moment does not touch this case.

Reply:
or at least the change should originate from someone else who has done real world testing.

Reply:
@ndyck14 If you don't feel comfortable changing snoop config, it's ok. No worries. Our team members can update this later. 😃 

My intention is to help share our experience from integrating and deploying Thread in different Nest/Google products 
and point out which configs can be useful in different scenarios.

---------

> seems this then become a question of how much un-acked traffic is travelling off the mesh. That's the only case that would create a potentially negative scenario from having the cache_snoop entries higher (i don't believe any of this matters if devices on mesh < max_cache).

Note that frames are acked at every hop at Mac layer. If a frame is sent across multiple hops over Thread mesh to a BR, the BR will not sent an ack to the original sender.

As I said, Thread network should behave fine even if snoop optimization is not used (i.e., setting the `MAX_SNOOP_ENTRIES` to zero). Using the snoop optimization will help reduce the delay in the (first) interaction (request response exchange) originating from a Thread device to an off-mesh entity (server). With snoop optimization when BR receives the response from server to forward within mesh, it does not need to perform address query for the mesh destination and can use the snooped cache entry.

Reply:
thanks @abtink , I definitely appreciate the suggestions. If you guys can add a separate PR, that would be preferrable. We'll definitely circle back to this once we have feature sets that reach out off-mesh proactively, which should be around Duckhorn timeframe.